itou/www/apply/tests/tests_list.py

```diff
@@ -126,6 +126,40 @@ def test_list_for_job_seeker_view_filtered_by_state(self):
         for application in applications:
             self.assertEqual(application.state, expected_state)
 
+    def test_list_for_job_seeker_view_filtered_by_dates(self):
+        """
+        Provide a list of job applications sent by a job seeker
+        and filtered by dates
+        """
+        now = timezone.now()
+
+        for diff_day in [7, 5, 3, 0]:
+            JobApplicationSentByJobSeekerFactory(
+                created_at=now - timezone.timedelta(days=diff_day), job_seeker=self.maggie
+            )
+
+        self.client.login(username=self.maggie.email, password=DEFAULT_PASSWORD)
+
+        date_format = DuetDatePickerWidget.INPUT_DATE_FORMAT
+
+        start_date = now - timezone.timedelta(days=5)
+        end_date = now - timezone.timedelta(days=1)
+        query = urlencode(
+            {
+                "start_date": timezone.localdate(start_date).strftime(date_format),
+                "end_date": timezone.localdate(end_date).strftime(date_format),
+            }
+        )
+        url = f"{self.job_seeker_base_url}?{query}"
+        response = self.client.get(url)
+        applications = response.context["job_applications_page"].object_list
+
+        self.assertEqual(len(applications), 2)
+
+        for application in applications:
```

Comment:
eviter les for dans les tests, et si tu as deux objets plutot tester l'indice puis le 1. Comme ça quand le test casse on sait tout de suite lequel des deux pose probleme.

si on a N objets, utiliser `subTest`.

Reply:
J'ai supprimé les `for`.